File: python/hello_world.py
"""Hello World application entry point.

Usage: python hello_world.py [language] [name]
  language: en (default), es, fr, de
  name: World (default)
"""

import sys

from src.greeter import Greeter


def main() -> None:
    language = sys.argv[1] if len(sys.argv) > 1 else "en"
    name = sys.argv[2] if len(sys.argv) > 2 else "World"

    greeter = Greeter(language)
    print(greeter.greet(name))
    print(f"Language: {greeter.language}")


if __name__ == "__main__":
    main()

File: java/src/main/java/com/example/HelloWorld.java
package com.example;

/**
 * Simple Hello World application demonstrating a multi-class Java project.
 *
 * <p>Usage: java -jar hello-world.jar [language] [name]
 *   language: en (default), es, fr, de
 *   name: World (default)
 */
public class HelloWorld {

    public static void main(String[] args) {
        String language = args.length > 0 ? args[0] : "en";
        String name = args.length > 1 ? args[1] : "World";

        Greeter greeter = new Greeter(language);
        System.out.println(greeter.greet(name));
        System.out.printf("Language: %s%n", greeter.getLanguage());
    }
}
